google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryOptions.java

```diff
@@ -61,6 +64,15 @@ private Builder(BigQueryOptions options) {
     }
 
     @Override
+    public Builder setTransportOptions(TransportOptions transportOptions) {
```

Comment:
Another possibility is to make `setTransportOptions` protected in `ServiceOptions` and make the signature of this method `public Builder setTransportOptions(HttpTransportOptions transportOptions)`. This can be changed to accept `TransportOptions` in future without breaking.

This doesn't work if someone holding a `ServiceOptions<?, ?, ?>` object needs to set the transport.